packages/eos-components/src/components/ChannelLogo.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <div
+    class="thumbnail"
+    :class="size"
+    :style="thumbnailStyles"
+  >
+  </div>
+</template>
+
+<script>
+  export default {
+    name: 'ChannelLogo',
+    props: {
+      channel: Object,
+      size: {
+        type: String,
+        default: 'sm',
+        validator: (value) => {
+          // The value must match one of these strings
+          return ['sm', 'md', 'lg', 'xl'].indexOf(value) !== -1;
+        },
+      },
+    },
+    data() {
+      return {
+        hasTransparentBg: false,
+      };
+    },
+    computed: {
+      thumbnailStyles() {
+        const styles = {
+          backgroundColor: 'white',
+          backgroundImage: `url(${this.channel.thumbnail})`,
+          backgroundSize: '100% auto',
+          backgroundPosition: 'center center',
+          backgroundRepeat: 'no-repeat',
+        };
+
+        if (this.hasTransparentBg) {
+          styles.backgroundSize = '80% auto';
+        }
+
+        return styles;
+      },
+    },
+    updated() {
+      this.calculateTransparency();
+    },
+    mounted() {
+      this.calculateTransparency();
+    },
+    methods: {
+      calculateTransparency() {
+        const canvas = document.createElement('canvas');
+        const ctx = canvas.getContext('2d');
+        const img = new Image();
+        img.onload = () => {
+          ctx.drawImage(img, 0, 0);
+          const { data } = ctx.getImageData(0, 0, 1, 1);
+          const [,,, alpha] = data;
+          this.hasTransparentBg = (alpha === 0);
+        };
+        img.src = this.channel.thumbnail;
+      },
```

Comment:
The logic looks great. But I thought we talked about not doing this for performance reasons. Also I'm not sure if this 80% resize will work for all the icons. In any case since you already implemented it we can merge it and see in the online instance. I'm really against fixing assets with code.